torch/ao/quantization/experimental/apot_utils.py

```diff
@@ -5,52 +5,64 @@
 
 import math
 
+from typing import Dict
+float_to_apot_cache: Dict[float, int] = dict()
+apot_to_float_cache: Dict[int, float] = dict()
+quant_dequant_cache: Dict[float, float] = dict()
+
 r"""Converts floating point input into APoT number
     based on quantization levels
 """
 def float_to_apot(x, levels, indices, alpha):
-    # clip values based on alpha
-    if x < -alpha:
-        return -alpha
-    elif x > alpha:
-        return alpha
-
-    levels_lst = list(levels)
-    indices_lst = list(indices)
+    if round(x, 4) in float_to_apot_cache:
+        return float_to_apot_cache[round(x, 4)]
```

Comment:
wait, doesn't this cache depend on the quantization parameters? i.e. if we change `levels`, 'indicies', 'alpha' this will break right?

Reply:
oh yes, i was running each instance of quantization parameters independently so it wouldn't be an issue but that would be an issue if they were run together -- will fix!